sci-libs/primegen/primegen-0.97.recipe

```diff
@@ -0,0 +1,81 @@
+SUMMARY="Fast C/C++ prime number generator library"
+DESCRIPTION="primegen is a small, fast library to generate prime numbers \
+in order. It generates the 50847534 primes up to 1000000000 in just 8 \
+seconds on a Pentium II-350; it prints them in decimal in just 35 seconds. \
+primegen can generate primes up to 1000000000000000, although it is not \
+optimized for primes past 32 bits. It uses the Sieve of Atkin instead of the \
+traditional Sieve of Eratosthenes."
+HOMEPAGE="https://cr.yp.to/primegen.html"
+COPYRIGHT="1999, D. J. Bernstein"
+LICENSE="Public Domain"
+REVISION="2"
+SOURCE_URI="https://cr.yp.to/primegen/primegen-$portVersion.tar.gz"
+CHECKSUM_SHA256="54285baf8eed9e421ff2220a2112d38cfb20c1ebef6014ef3f0004c22c95f40d"
+SOURCE_DIR="primegen-$portVersion"
+PATCHES="primegen-$portVersion.patchset"
+
+ARCHITECTURES="?x86_gcc2 x86 x86_64"
+
+PROVIDES="
+	primegen = $portVersion
+	cmd:primegaps = $portVersion
+	cmd:primes = $portVersion
+	cmd:primespeed = $portVersion
+	"
+REQUIRES="
+		haiku
+		"
+
+PROVIDES_devel="
+	primegen_devel = $portVersion
+	devel:libprimegen
+	"
+REQUIRES_devel="
+	primegen == $portVersion base
+	"
+
+BUILD_PREREQUIRES="
+		cmd:make
+		cmd:gcc
+		cmd:nroff
+		"
+
+BUILD_REQUIRES="
+		haiku_devel
+		"
+
+BUILD()
+{
+	make
+}
+
+INSTALL()
+{
+	install -d $binDir $docDir $manDir/man1 $manDir/man3
+	install -d $includeDir/primegen $developLibDir $libDir
+	install -t $binDir primegaps primes primespeed
+	install -t $docDir README
+	install -t $manDir/man1 primegaps.1 primes.1 primespeed.1
+	install -t $manDir/man3 primegen.3
+	install -t $includeDir/primegen primegen.h uint32.h uint64.h
+	install -T primegen.a $developLibDir/libprimegen.a
+	install -T primegen.so $developLibDir/libprimegen.so
+	install -T primegen.so $libDir/libprimegen.so
```

Comment:
I think it would be better to replace lines 62 and 63 by:
```
	install -T primegen.so $libDir/libprimegen.so
	prepareInstalledDevelLib libprimegen
	fixPkgconfig
```
but I haven't checked yet.

Reply:
This doesn't affect the final output, so I made those changes.

Reply:
Hmm, I think there is a tiny difference. When we use `prepareInstalledDevelLib` we get a symbolic link (for develop/lib/libprimegen.so) in the primegen_devel package that points to the real file (lib/libprimegen.so) in the primegen package.
So I would recommend using `prepareInstalledDevelLib`.